rust/sealevel/client/src/router.rs

```diff
@@ -125,7 +126,19 @@ impl ChainMetadata {
     }
 
     pub fn domain_id(&self) -> u32 {
-        self.domain_id.unwrap_or(self.chain_id)
+        self.domain_id.unwrap_or_else(|| {
+            // Try to parse as a number, otherwise panic, as the domain ID must
+            // be specified if the chain id is not a number.
+            self.chain_id
+                .as_u64()
+                .and_then(|v| v.try_into().ok())
+                .unwrap_or_else(|| {
+                    panic!(
+                        "Invalid chain_id cannot be used as the domain_id: {:?}",
```

Comment:
nit - as a CLI user I'd find something like this easier to understand: "`domain_id` is undefined, nor could it be inferred from `chain_id` due to a parsing error"